demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTableZipExporter.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Logic\Statement\AssessmentTableExporter;
+
+use demosplan\DemosPlanCoreBundle\Entity\File;
+use Exception;
+use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
+
+class AssessmentTableZipExporter extends AssessmentTableXlsExporter
+{
+    protected array $supportedTypes = ['zip'];
+
+    /**
+     * @throws Exception
+     */
+    public function __invoke(array $parameters): array
+    {
+        $xlsxArray = parent::__invoke($parameters);
+
+        $statementAttachments = $this->getAttachmentsOfStatements($parameters['items']);
```

Comment:
is key items always defined?

Reply:
I think items was generally wrong in place. Instead, the xlsx exporter now returns the IDs. And the key statementIds should actually always be set. see [c7a99c1](https://github.com/demos-europe/demosplan-core/pull/2337/commits/c7a99c1f6a3c496b92e7067c7201910b11b364c2)
If not, an error would have to be thrown that would be caught in the controller. The user receives an error message and the Zip export is aborted.

Reply:
also adjusted with: [3e35f11](https://github.com/demos-europe/demosplan-core/pull/2337/commits/3e35f1155dc80c3ccefb7901eb1f581f47a9258b)